### PR TITLE
Allow reporters to update display_name if insights_id isn't set

### PR DESCRIPTION
**Pull request by kruai**
## Overview

This PR is being created to address [RHCLOUD-13105](https://issues.redhat.com/browse/RHCLOUD-13105).
It allows reporters to update a Host's `display_name` if the Insights ID hasn't been set.
This functionality is required to support rhsm-conduit's hypervisor/vsphere implementation.

## Secure Coding Practices Checklist GitHub Link

- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist

- [x] Input Validation
- [x] Access Control
- [x] General Coding Practices

### CODE REVIEW

**Review by tahmidefaz — APPROVED** [review]
LGTM :+1:

**Review by ryandillinfelton — APPROVED** [review]
This all looks good :+1:

File: tests/test_models.py [hunk 1]
```diff
@@ -110,16 +110,36 @@ def test_update_existing_host_fix_display_name_using_id(db_create_host):
...
+
+    # Attempt to update the display name from Satellite reporter.
+    # Should update because Insights ID isn't set.
+    expected_fqdn = "different.domain1.com"
+    new_display_name = "change_me"
+    input_host = Host({"fqdn": expected_fqdn}, display_name=new_display_name, reporter="yupana", stale_timestamp=now())
+    existing_host.update(input_host)
+
+    # assert display name has changed
+    assert existing_host.display_name == new_display_name
```
Comment (thearifismail) [inline tests/test_models.py:142]: Should assert for `expected_fqdn`

Comment (kruai) [reply tests/test_models.py:142]: Do you think that's necessary? This test exists to validate `display_name` functionality. If you think it's important to the test, though, I can add it

Comment (thearifismail) [reply tests/test_models.py:142]: it was just for confirmation to make sure but okay to leave it out.

**Review by thearifismail — APPROVED** [review]

**Comment by kruai** [comment]
I asked @lphiri if the Satellite issue which prompted the initial workaround has been resolved. He's not sure yet, but is looking into it. Because it's taking time to research, I propose we merge this and create a follow-up Jira to remove the display_name overrides (and related tests) once he confirms that the Satellite issue is resolved.

**Pull request merged**